node/src/neighborhood/gossip_acceptor.rs

```diff
@@ -1518,57 +1563,89 @@ mod tests {
+            .unwrap()
+            .node_addr_opt()
+            .unwrap();
+        assert_eq!(db_node_node_addr, new_addr);
+        let make_expected = |index: usize| RemoveStreamMsg {
+            local_addr: SocketAddr::from(db.root().node_addr_opt().unwrap()),
+            peer_addr: SocketAddr::new(old_addr.ip_addr(), old_addr.ports()[index]),
+            stream_type: RemovedStreamType::Clandestine,
+            sub: dispatcher_sub.clone(),
+        };
         let stream_handler_pool_recording = stream_handler_pool_recording_arc.lock().unwrap();
         assert_eq!(
             stream_handler_pool_recording.get_record::<RemoveStreamMsg>(0),
-            &RemoveStreamMsg {
-                local_addr: SocketAddr::from (db.root().node_addr_opt().unwrap()),
-                peer_addr: SocketAddr::new(old_addr.ip_addr(), old_addr.ports()[0]),
-                stream_type: RemovedStreamType::Clandestine,
-                sub: dispatcher_sub.clone()
-            }
+            &make_expected(0)
         );
         assert_eq!(
             stream_handler_pool_recording.get_record::<RemoveStreamMsg>(1),
-            &RemoveStreamMsg {
-                local_addr: SocketAddr::from (db.root().node_addr_opt().unwrap()),
-                peer_addr: SocketAddr::new(old_addr.ip_addr(), old_addr.ports()[1]),
-                stream_type: RemovedStreamType::Clandestine,
-                sub: dispatcher_sub
-            }
+            &make_expected(1)
+        );
+    }
+
+    #[test]
+    fn ipchange_accepting_connections_but_without_node_addr_is_rejected() {
+        let old_addr = NodeAddr::new(&IpAddr::from_str("1.2.3.4").unwrap(), &vec![2000, 2001]);
+        let new_addr = NodeAddr::new(&IpAddr::from_str("2.3.4.5").unwrap(), &vec![2000, 2001]);
+        let gossip_source = SocketAddr::from(new_addr.clone());
+        let (new_ipchange_node, src_db, mut dest_db) = make_ipchange_databases(
+            2345,
+            Some(old_addr.clone()),
+            Some(new_addr.clone()),
+            Mode::Standard,
+        );
+        let gossip = GossipBuilder::new(&src_db)
+            .node(new_ipchange_node.public_key(), false)
+            .build();
+        let agrs_vec: Vec<AccessibleGossipRecord> = gossip.try_into().unwrap();
+        let (stream_handler_pool, _, _) = make_recorder();
+        let (dispatcher, _, _) = make_recorder();
+        let stream_handler_pool_sub = stream_handler_pool.start().recipient();
+        let dispatcher_sub = dispatcher.start().recipient();
+        let subject =
+            IpChangeHandler::new(stream_handler_pool_sub, dispatcher_sub, Logger::new("test"));
+
+        let result = subject.qualifies(&mut dest_db, agrs_vec.as_slice(), gossip_source);
+
+        assert_eq!(
+            result,
+            Qualification::Malformed(
+                "IpChange Gossip: Node AgMEBQ accepts connections but provides no NodeAddr"
+                    .to_string()
+            ),
```

Comment:
Is there any logging placed near to this event (not necessarily in the tested function) so that we can later do an analysis if something like this has come about?

Reply:
I think so. All the malformations are logged, and they'll be right next to the gossip diagram, if TRACE logging is on.